### PR TITLE
Collections cleanup

### PR DESCRIPTION
**Pull request by Gankra**
This is 99% burning ints to the ground, but I also got rid of useless annotations or made code more "idiomatic" as I went along. Mostly changes in tests.

### CODE REVIEW

File: src/libcollections/slice.rs [hunk 1]
```diff
@@ -812,22 +812,22 @@ impl<T> SliceExt for [T] {
     }
 
     #[inline]
-    fn slice<'a>(&'a self, start: uint, end: uint) -> &'a [T] {
+    fn slice<'a>(&'a self, start: usize, end: usize) -> &'a [T] {
```
Comment (apasel422) [inline src/libcollections/slice.rs]: I think these lifetimes (and the corresponding ones on similar methods) can be elided.

Comment (Gankro) [inline src/libcollections/slice.rs]: Ah yeah, I wasn't checking for that.

**Comment by Gankro** [comment]
@apasel422 I did a quick pass to remove a bunch of useless lifetimes.

**Comment by apasel422** [comment]
@Gankro Awesome. Looks good to me!

**Comment by alexcrichton** [comment]
Nice work @Gankro, thanks!

Can you  also remove the `int_uint` feature from `libcollections/lib.rs` while you're at it? Otherwise r=me

**Comment by Gankro** [comment]
Oh hey, that shows me all the places I missed. Sweet. New commit with all the missed stuff.

**Comment by Gankro** [comment]
@bors r=alexcrichton 15fb06 p=1

(want to fast track since it's merge-conflicty)

**Comment by Gankro** [comment]
@bors p=0

@Manishearth's rollup'll handle this

**Comment by pnkfelix** [comment]
Hmm, I think removing the `int_uint` feature may be what caused the latest rollup attempt on #21969 to fail, see : http://buildbot.rust-lang.org/builders/auto-mac-64-opt/builds/3485/steps/test/logs/stdio

**Comment by Manishearth** [comment]
Wait, I fixed those errors. Hm

**Comment by Manishearth** [comment]
Ah, I hadn't run doctests

**Comment by Gankro** [comment]
Oh whoops, I killed my tests before doctests ran.

**Comment by Gankro** [comment]
Merged in rollup with some extra patches

**Pull request closed without merging**